lib/topological_inventory/mock_source/collector.rb

```diff
@@ -20,10 +20,35 @@ def initialize(source, config, amounts)
       end
 
       def collect!
-        if ::Settings.multithreading == :on
+        do_loop = true
+        while do_loop
```

Comment:
You could just use a `loop do` here and after the refresh `break unless ::Settings.refresh_mode == :full_refresh`

Reply:
I kept if/else rather than if + unless, but I get the point.